lang/java/avro/src/main/java/org/apache/avro/io/parsing/Symbol.java

```diff
@@ -490,10 +511,20 @@ public static EnumAdjustAction enumAdjustAction(int rsymCount, Object[] adj) {
   }
 
   public static class EnumAdjustAction extends IntCheckAction {
+    public final boolean noAdjustments;
     public final Object[] adjustments;
     @Deprecated public EnumAdjustAction(int rsymCount, Object[] adjustments) {
       super(rsymCount);
       this.adjustments = adjustments;
+      boolean noAdj = true;
+      if (adjustments != null) {
+        int count = Math.min(rsymCount, adjustments.length);
+        noAdj = (adjustments.length <= rsymCount);
+        for (int i = 0; noAdj && i < count; i++)
+          noAdj &= ((adjustments[i] instanceof Integer)
```

Comment:
We can break out of the loop when we see first `false`.

Reply:
I think you highlighted this as an illustration, rather than as a "todo" item, so I'm resolving the comment.